lib/ansible/modules/source_control/gitlab_deploy_key.py

```diff
@@ -9,35 +7,46 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-
 ANSIBLE_METADATA = {'metadata_version': '1.1',
                     'status': ['preview'],
                     'supported_by': 'community'}
 
-
 DOCUMENTATION = '''
----
 module: gitlab_deploy_key
 short_description: Manages GitLab project deploy keys.
 description:
      - Adds, updates and removes project deploy keys
 version_added: "2.6"
 options:
-  api_url:
+  server_url:
     description:
-      - GitLab API url, e.g. https://gitlab.example.com/api
+      - Url of Gitlab server, with protocol (http or https).
     required: true
-  access_token:
+    version_added: "2.8"
+  verify_ssl:
     description:
-      - The oauth key provided by GitLab. One of access_token or private_token is required. See https://docs.gitlab.com/ee/api/oauth2.html
-    required: false
-  private_token:
+      - When using https if SSL certificate needs to be verified.
+    type: bool
+    default: 'yes'
+    aliases:
+      - validate_certs
+      - enable_ssl_verification
+    version_added: "2.8"
+  login_user:
     description:
-      - Personal access token to use. One of private_token or access_token is required. See https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html
-    required: false
+      - Gitlab user name.
+    version_added: "2.8"
+  login_password:
+    description:
+      - Gitlab password for login_user
+    version_added: "2.8"
+  login_token:
+    description:
+      - Gitlab token for logging in.
+    version_added: "2.8"
   project:
     description:
-      - Numeric project id or name of project in the form of group/name
+      - Full path of project in the form of group/name
```

Comment:
It seems like python-gitlab [supports numeric ids](https://python-gitlab.readthedocs.io/en/stable/gl_objects/projects.html), any reason to limit the docs here?

Reply:
I will create a function in gitlab library to allow id again